src/viam/sessions_client.py

```diff
@@ -103,66 +109,58 @@ async def metadata(self) -> _MetadataLike:
         if response.heartbeat_window is None:
             raise GRPCError(status=Status.INTERNAL, message="Expected heartbeat window in response to start session")
 
-        self._supported.value = _SupportedState.TRUE
-        self._heartbeat_interval = response.heartbeat_window.ToTimedelta()
-        self._current_id = response.id
+        with self._lock:
+            self._supported = _SupportedState.TRUE
+            self._heartbeat_interval = response.heartbeat_window.ToTimedelta()
+            self._current_id = response.id
 
         # tick once to ensure heartbeats are supported
-        await _heartbeat_tick(self.client, self._current_id, self._supported)
+        await self._heartbeat_tick(self.client)
+
+        with self._lock:
+            if self._supported == _SupportedState.TRUE:
+                # We send heartbeats slightly faster than the interval window to
+                # ensure that we don't fall outside of it and expire the session.
+                wait = self._heartbeat_interval.total_seconds() / 5
+
+                self._thread = Thread(
+                    name="heartbeat-thread",
+                    target=asyncio.run,
+                    args=(self._heartbeat_process(wait),),
+                    daemon=True,
+                )
+                self._thread.start()
```

Comment:
Q: is it possible to get here with an active thread? (e.g. `self._thread is not None`). If so, do we need to do any cleanup?

Reply:
I think it is possible, added a reset if so